ocs_ci/ocs/cnv/virtual_machine_instance.py

```diff
@@ -0,0 +1,195 @@
+"""
+Virtual machine instance class
+"""
+import logging
+
+from ocs_ci.ocs.ocp import OCP
+from ocs_ci.ocs.cnv.virtctl import Virtctl
+from ocs_ci.ocs.resources.pod import Pod
+from ocs_ci.ocs.resources.pod import get_pods_having_label
+from ocs_ci.utility.utils import TimeoutSampler
+from ocs_ci.ocs import constants
+
+
+logger = logging.getLogger(__name__)
+
+
+class VirtualMachineInstance:
```

Comment:
@PrasadDesala I am not able to clearly understand the difference between VirtualMachine class and VirtualMachineInstance class. To me it looks mostly similar (maybe I lack complete understanding of how CNV + ODF works), operations looks similar and VirtualMachine class functions appears as if they are operating on actual instances. Could you please clarify.

Reply:
You are right! VM defines the desired state and configuration and it is not associated with a specific pod or process, whereas VMI represents an instance of a running virtual machine.
A few  of the operation calls like delete, start, pause look to be same but it's behavior and the internal calls will be different from each other.
For instance,if we want to stop a VM,  the VM will go to stopped state only after the VMI deletes its associated virt_launcher pod and vmi resource. We need to have separate functions as each maintains it's own crds.